src/frontends/tensorflow/src/graph_iterator_saved_model.hpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <map>
+
+#include "graph_iterator_proto.hpp"
+#include "openvino/util/file_util.hpp"
+#include "saved_model.pb.h"
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+
+struct VIBlock;
+
+template <typename T>
+std::basic_string<T> getSMName() {}
+template <typename T>
+std::basic_string<T> getVIName() {}
+
+template <>
+std::basic_string<char> getSMName<char>();
+template <>
+std::basic_string<char> getVIName<char>();
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+template <>
+std::basic_string<wchar_t> getSMName<wchar_t>();
+template <>
+std::basic_string<wchar_t> getVIName<wchar_t>();
+#endif
+
+// Stores information about variables index
+class SMVariablesIndex {
+    int32_t totalShards;
+    std::map<std::string, std::vector<char>> varIndex;
```

Comment:
please use `unordered_map`

Reply:
Hmmm… reason? Do you think it is performance bottleneck? Why? It will increase memory usage with non-measurable performance profit in exact this case.
Do you think it should be used only here or for both maps? Why?

Reply:
My motivation: because it provides worse memory usage with non-capturable performance benefits in current implementation.